### PR TITLE
adding support for custom location of Roslyn binaries

### PR DESCRIPTION
**Pull request by Jinhuafei**

### CODE REVIEW

File: src/Microsoft.CodeDom.Providers.DotNetCompilerPlatform/Util/CompilationSettings.cs [hunk 1]
```diff
@@ -56,8 +59,17 @@ internal static class CompilationSettingsHelper {
                 ttl = DefaultCompilerServerTTLInDevEnvironment;
             }
 
-            _csc = new CompilerSettings(CompilerFullPath(@"bin\roslyn\csc.exe"), ttl);
-            _vb = new CompilerSettings(CompilerFullPath(@"bin\roslyn\vbc.exe"), ttl);
+            var customRoslynCompilerLocation = Environment.GetEnvironmentVariable(CustomRoslynCompilerLocation, EnvironmentVariableTarget.Process);
+            if(customRoslynCompilerLocation != null)
+            {
+                _csc = new CompilerSettings($"{customRoslynCompilerLocation}\\csc.exe", ttl);
```
Comment (StephenMolloy) [inline src/Microsoft.CodeDom.Providers.DotNetCompilerPlatform/Util/CompilationSettings.cs]: Is it safe to assume this C# 6 syntax is ok everywhere? I assume it gets compiled away and all. Just asking.

Comment (Jinhuafei) [inline src/Microsoft.CodeDom.Providers.DotNetCompilerPlatform/Util/CompilationSettings.cs]: Msbuild 14 supports C# 6 and it was shipped 2 years ago, so it should be safe to assume that. And it's VS2015 solution, I believe msbuild was updated in update 2 or 3 to support C#6.

**Comment by niemyjski** [comment]
How would this work? You'd have to set an environment variable in code scoped to the process? It would be good to just look in the root bin folder or pass in a folder path. So if you installed the nuget package everything was copied to that folder location like it's always been.

**Comment by Jinhuafei** [comment]
This is for server scenario. Basically you add a machine wide environment variable and point it to a Rolsyn compiler location. This is useful for the hosting scenario, where it's not allowed to upload exe files. Also, the server owner can just manage one copy of Roslyn bits on the machine.

The reason we can't copy Roslyn bits directly to the bin folder is Roslyn compilers depend on many assemblies released through Nuget. It's very possible they will overwrite the project references or be overwritten.

**Pull request merged**